test/e2e/resize_nodes.go

```diff
@@ -434,9 +410,17 @@ var _ = Describe("Nodes", func() {
 			if err := waitForGroupSize(testContext.CloudConfig.NumNodes); err != nil {
 				Failf("Couldn't restore the original node instance group size: %v", err)
 			}
-			if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes, 10*time.Minute); err != nil {
+			if err := waitForClusterSize(framework.Client, testContext.CloudConfig.NumNodes, 10*time.Minute); err != nil {
 				Failf("Couldn't restore the original cluster size: %v", err)
 			}
+			// Many e2e tests assume that the cluster is fully healthy before they start.  Wait until
+			// the cluster is restored to health
+			By("waiting for system pods to successfully restart")
+			pods, err := framework.Client.Pods("kube-system").List(labels.Everything(), fields.Everything())
```

Comment:
The trouble is that when you kill a node, and add a new one things get rescheduled, so things like influx and kube-dns aren't running yet.  And subsequent tests fail because they expect those containers to already be up and running.